Snake.h

```diff
@@ -0,0 +1,66 @@
+#ifndef SNAKE_H
+#define SNAKE_H
+
+#include <cassert>
+
+class Snake
+{
+public:
+    enum Direction
```

Comment:
Use class enum instead (also read about the difference between enum and class enum)